mdc_rpt_physical_worksheet/report/report_xlsx.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, tools, models, _
+import base64
+import tempfile
+import datetime, pytz
+
+
+def formats():
+    # Python dictionary with cells formats and styles for XLSX reports
+    dic_formats={}
+    dic_formats["title"] = {'bold': True, 'size': 30}
+    dic_formats["filter"] = {'bold': True,
+                                  'font_color': '#004080',
+                                  'size': 16}
+    dic_formats["filter_date"] = {'bold': True,
+                             'font_color': '#004080',
+                             'size': 16,
+                             'num_format': '[$-x-sysdate]dddd, mmmm dd, aaaa'}
+    dic_formats["header"] = {'bold': True,
+                             'font_color': '#004080',
+                             'bottom': True,
+                             'top': True,
+                             'text_wrap': True,
+                             'align': 'center',
+                             'size': 12}
+    dic_formats["header_ind"] = {'bold': True,
+                             'font_color': '#004080',
+                             'bg_color': '#BBDDBB',
+                             'bottom': True,
+                             'top': True,
+                             'text_wrap': True,
+                             'align': 'center',
+                             'size': 12}
+    dic_formats["data"] = {'align': 'center',
+                             'num_format': '0'}
+    dic_formats["dataL"] = {'align': 'left',
+                           'num_format': '0'}
+    dic_formats["data2d"] = {'align': 'center',
+                             'num_format': '0.00'}
+    dic_formats["dataF"] = {'align': 'center',
+                             'bg_color': '#BBDDBB',
+                             'num_format': '0'}
+    dic_formats["dataY"] = {'align': 'center',
+                             'bg_color': '#FFFF00',
+                             'num_format': '0'}
+    dic_formats["data2dY"] = {'align': 'center',
+                             'bg_color': '#FFFF00',
+                             'num_format': '0.00'}
+    return dic_formats
+
+
+class ReportRptPhysicalWorksheetXlsx(models.AbstractModel):
+    _name = 'report.mdc.rpt_physical_worksheet'
+    _inherit = 'report.report_xlsx.abstract'
+
+    def convert_UTC_to_TZ(self, utc_dt):
+        # this function is based in this two links
+        # https://www.odoo.com/es_ES/forum/ayuda-1/question/compute-datetime-field-from-date-how-to-deal-with-timezones-135727
+        # https://stackoverflow.com/questions/79797/how-to-convert-local-time-string-to-utc
+        context_tz = pytz.timezone(self.env.context.get('tz') or self.env.user.tz or 'Europe/Madrid')
+        # Convert to current user time zone
+        dt_local = context_tz.localize(utc_dt, is_dst=None)
+        diffH = int(str(dt_local)[19:22])
+        local_dt = utc_dt + datetime.timedelta(hours=diffH)
+        return local_dt
```

Comment:
Need to review, I'll take a look:

> `context_tz = pytz.timezone(self.env.context.get('tz') or self.env.user.tz or 'Europe/Madrid')`

I don't know if environment will always provide the timezone; the user usually has it undefined. So, the last option should be a module configuration parameter (such as "Pyshical worksheet report Default timezone").

> `diffH = int(str(dt_local)[19:22])`

This is the weakest point of the code, I'll made some tests or look for an alternative

Reply:
Tested, these are my conclusions:
- Algorithm to adapt timezone seems ok in general.
- `self.env.context.get('tz')` actually gets the same value than `self.env.user.tz`, and can be both `False`. So, a general parameter configuration is mandatory.
- The current code should fail in two situations: (1) since the timezone adaptations are only applied to time value, sometimes the date value could be wrong (e.g. a worksheet around midnight), and (2) less important, the hour offset should include 1/2 hour fragments, e.g. India +05:30 XD

I'd develop the general configuration parameter, @lmiguens-solvos could you fix (1)?

Reply:
Finally I found an alternative and clearer code. Could you check?

```suggestion
    def convert_UTC_to_TZ(self, utc_dt):
        context_tz = pytz.timezone(self.env.context.get('tz') or self.env.user.tz or 'Europe/Madrid')
        return utc_dt + datetime.timedelta(seconds=context_tz.utcoffset(utc_dt).total_seconds())
```

Reply:
It works perfectly

Reply:
Still missing (1) issue with datetimes around midnight